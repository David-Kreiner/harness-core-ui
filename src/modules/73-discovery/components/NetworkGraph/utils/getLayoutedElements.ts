/*
 * Copyright 2023 Harness Inc. All rights reserved.
 * Use of this source code is governed by the PolyForm Shield 1.0.0 license
 * that can be found in the licenses directory at the root of this repository, also available at
 * https://polyformproject.org/wp-content/uploads/2020/06/PolyForm-Shield-1.0.0.txt.
 */

import ELK, { ElkExtendedEdge, ElkNode } from 'elkjs/lib/elk.bundled'
import type { Edge } from 'reactflow'
import type { LayoutedGraph } from '../types'
import { NodeTypes } from '../constants'

const elk = new ELK()

const options = {
  'elk.algorithm': 'stress'
}

export async function getLayoutedElements({ nodes, edges }: LayoutedGraph): Promise<LayoutedGraph> {
  // Generate elk graph node from reactflow node structure
  // Get the group nodes
  const groupNodes = nodes.filter(node => node.type === NodeTypes.Group)

  // Create parent elk nodes with it's children and edges
  const parentNodes = groupNodes.map(groupNode => {
    return {
      ...groupNode,
      children: nodes
        .filter(node => node.parentNode === groupNode.id)
        .map(node => ({
          ...node,
          // Adjust the target and source handle positions based on the layout
          // direction.
          targetPosition: 'top',
          sourcePosition: 'bottom',

          // Hardcode a width and height for elk to use when layouting.
          width: 100,
          height: 100
        })),
      edges: edges.filter(
        edge => /* istanbul ignore next */ edge.data?.parentNode === groupNode.id
      ) as unknown as ElkExtendedEdge[]
    }
  })

  const graph: ElkNode = {
    id: 'root',
    layoutOptions: options,
    children: parentNodes.map(node => ({
      ...node,
      // Hardcode a width and height for elk to use when layouting.
      width: 0,
      height: 0
    })),
    edges: edges as unknown as ElkExtendedEdge[]
  }

  const elkGraph = await elk.layout(graph)

  /* istanbul ignore next */
  const flattenedNodes = elkGraph?.children?.map(node => ({
    ...node,
    // React Flow expects a position property on the node instead of `x`
    // and `y` fields.
    position: { x: node.x, y: node.y }
  }))

  /* istanbul ignore next */
  elkGraph?.children?.map(node => {
    if (node.children) {
      /* istanbul ignore next */
      flattenedNodes?.push(
        ...node.children.map(n => ({
          ...n,
          // React Flow expects a position property on the node instead of `x`
          // and `y` fields.
          position: { x: n.x, y: n.y }
        }))
      )
    }
  })

  const layoutedGraph = {
    nodes: flattenedNodes,
    edges: elkGraph.edges as unknown as Edge<unknown>[]
  } as LayoutedGraph

  return layoutedGraph
}
