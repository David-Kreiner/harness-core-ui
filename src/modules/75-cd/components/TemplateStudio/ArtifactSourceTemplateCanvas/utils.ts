import type { ArtifactType } from '@pipeline/components/ArtifactsSelection/ArtifactInterface'
import { ArtifactToConnectorMap } from '@pipeline/components/ArtifactsSelection/ArtifactHelper'

export interface Params {
  oldConnectorRef?: string
  newConnectorRef?: string
  artifactType: ArtifactType
}

export const isConnectorRefUpdated = (params: Params) => {
  const { oldConnectorRef, newConnectorRef, artifactType } = params
  const isConnectorRefApplicable = Boolean(ArtifactToConnectorMap[artifactType])

  return Boolean(isConnectorRefApplicable && oldConnectorRef && newConnectorRef && oldConnectorRef !== newConnectorRef)
}
