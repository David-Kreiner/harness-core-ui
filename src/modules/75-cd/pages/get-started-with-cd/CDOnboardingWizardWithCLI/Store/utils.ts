export const parseJSON = (str: string): object => {
  try {
    return JSON.parse(str)
  } catch (e) {
    return {}
  }
}
