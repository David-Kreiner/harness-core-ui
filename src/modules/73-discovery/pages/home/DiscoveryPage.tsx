/*
 * Copyright 2022 Harness Inc. All rights reserved.
 * Use of this source code is governed by the PolyForm Shield 1.0.0 license
 * that can be found in the licenses directory at the root of this repository, also available at
 * https://polyformproject.org/wp-content/uploads/2020/06/PolyForm-Shield-1.0.0.txt.
 */

import React, { useState } from 'react'
import { useParams } from 'react-router-dom'
import { Button, ButtonVariation, Container, ExpandingSearchInput, Layout } from '@harness/uicore'
import { Drawer, Position } from '@blueprintjs/core'
import { useDocumentTitle } from '@common/hooks/useDocumentTitle'
import { useStrings } from 'framework/strings'
import { Page } from '@common/exports'
import type { ModulePathParams, ProjectPathProps } from '@common/interfaces/RouteInterfaces'
import { NGBreadcrumbs } from '@common/components/NGBreadcrumbs/NGBreadcrumbs'
import { getLinkForAccountResources } from '@common/utils/BreadcrumbUtils'
import { Scope } from '@common/interfaces/SecretsInterface'
import ScopedTitle from '@common/components/Title/ScopedTitle'
import { ApiGetAgentResponse, useListAgent } from 'services/servicediscovery'
import RbacButton from '@rbac/components/Button/Button'
import DiscoveryAgentTable from '@discovery/components/DiscoveryAgentTable/DiscoveryAgentTable'
import { useQueryParams } from '@common/hooks'
import { useDefaultPaginationProps } from '@common/hooks/useDefaultPaginationProps'
import { DEFAULT_PAGE_SIZE, ServiceDiscoveryFilterParams } from '@discovery/interface/filters'
import EmptyStateDiscoveryAgent from './views/empty-state/EmptyStateDiscoveryAgent'
import CreateDAgent from './views/create-discovery-agent/CreateDAgent'
import css from './DiscoveryPage.module.scss'

const DiscoveryPage: React.FC = /* istanbul ignore next */ () => {
  const { accountId, orgIdentifier, projectIdentifier } = useParams<ProjectPathProps & ModulePathParams>()
  const { getString } = useStrings()
  const discoveryLabel = getString('common.discovery')
  const [search, setSearch] = useState('')
  const [isOpen, setDrawerOpen] = useState(false)
  useDocumentTitle(discoveryLabel)

  //States for pagination
  const { page, size } = useQueryParams<ServiceDiscoveryFilterParams>()

  const { data: discoveryAgentList, loading: discoveryAgentListLoading } = useListAgent({
    queryParams: {
      accountIdentifier: accountId,
      organizationIdentifier: orgIdentifier,
      projectIdentifier: projectIdentifier,
      search: search,
      all: false,
      page: page ? parseInt(page) : 0,
      limit: size ? parseInt(size) : DEFAULT_PAGE_SIZE
    }
  })

  const paginationProps = useDefaultPaginationProps({
    itemCount: discoveryAgentList?.page?.totalItems ?? 0,
    pageSize: size ? parseInt(size) : DEFAULT_PAGE_SIZE,
    pageCount: discoveryAgentList?.page?.totalPages ?? 1,
    pageIndex: page ? parseInt(page) : 0
  })

  const discoveryAgentListData: ApiGetAgentResponse[] = React.useMemo(
    () => discoveryAgentList?.items || [],
    [discoveryAgentList?.items]
  )

  return (
    <Container>
      <Page.Header
        breadcrumbs={
          <NGBreadcrumbs
            links={getLinkForAccountResources({ accountId, orgIdentifier, projectIdentifier, getString })}
          />
        }
        title={
          <ScopedTitle
            title={{
              [Scope.PROJECT]: discoveryLabel,
              [Scope.ORG]: discoveryLabel,
              [Scope.ACCOUNT]: discoveryLabel
            }}
          />
        }
      />

      <Page.Body loading={discoveryAgentListLoading}>
        {discoveryAgentList && discoveryAgentList.items && discoveryAgentList?.items?.length > 0 ? (
          <>
            <Page.SubHeader>
              <Layout.Horizontal flex={{ justifyContent: 'space-between' }} width={'100%'}>
                <Layout.Horizontal>
                  <RbacButton
                    text={getString('discovery.homepage.newDiscoveryAgentBtn')}
                    variation={ButtonVariation.PRIMARY}
                    icon="plus"
                    onClick={() => setDrawerOpen(true)}
                  />
                </Layout.Horizontal>
                <Container data-name="monitoredServiceSeachContainer">
                  <ExpandingSearchInput
                    width={250}
                    alwaysExpanded
                    throttle={500}
                    defaultValue={search}
                    onChange={value => setSearch(value)}
                    placeholder={getString('discovery.homepage.searchDiscoveryAgent')}
                  />
                </Container>
              </Layout.Horizontal>
            </Page.SubHeader>
            <Page.Body className={css.discoveryAgentTable}>
              <DiscoveryAgentTable listData={discoveryAgentListData} pagination={paginationProps} />
            </Page.Body>
          </>
        ) : (
          <EmptyStateDiscoveryAgent setDrawerOpen={setDrawerOpen} />
        )}
      </Page.Body>
      <Drawer position={Position.RIGHT} isOpen={isOpen} isCloseButtonShown={true} size={'86%'}>
        <Button
          minimal
          className={css.almostFullScreenCloseBtn}
          icon="cross"
          withoutBoxShadow
          onClick={() => setDrawerOpen(false)}
        />
        <CreateDAgent setDrawerOpen={setDrawerOpen} />
      </Drawer>
    </Container>
  )
}

export default DiscoveryPage
