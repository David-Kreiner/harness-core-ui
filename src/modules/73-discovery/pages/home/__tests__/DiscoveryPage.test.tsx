/*
 * Copyright 2021 Harness Inc. All rights reserved.
 * Use of this source code is governed by the PolyForm Shield 1.0.0 license
 * that can be found in the licenses directory at the root of this repository, also available at
 * https://polyformproject.org/wp-content/uploads/2020/06/PolyForm-Shield-1.0.0.txt.
 */

import React from 'react'
import { render } from '@testing-library/react'
import { TestWrapper } from '@common/utils/testUtils'
import routes from '@common/RouteDefinitions'
import * as servicediscovery from 'services/servicediscovery'
import { accountPathProps, modulePathProps, projectPathProps } from '@common/utils/routeUtils'
import { ApiGetAgentResponse, useListAgent } from 'services/servicediscovery'
import type { DrawerProps } from '../views/create-discovery-agent/CreateDAgent'
import DiscoveryPage from '../DiscoveryPage'

jest.mock('@discovery/pages/home/views/create-discovery-agent/CreateDAgent', () => ({
  ...jest.requireActual('@discovery/pages/home/views/create-discovery-agent/CreateDAgent'),
  __esModule: true,
  default: () => {
    return <div className={'create-d-agent'}>CreateDAgent</div>
  }
}))

jest.mock('@discovery/pages/home/views/empty-state/EmptyStateDiscoveryAgent', () => ({
  ...jest.requireActual('@discovery/pages/home/views/empty-state/EmptyStateDiscoveryAgent'),
  __esModule: true,
  default: (props: DrawerProps) => {
    return (
      <div className={'discovery-view-mock'}>
        <h1>Empty State Discovery Table</h1>
        <button onClick={() => props.setDrawerOpen(prev => !prev)}>Open Discovery Drawer</button>
      </div>
    )
  }
}))

jest.mock('@discovery/components/DiscoveryAgentTable/DiscoveryAgentTable', () => ({
  ...jest.requireActual('@discovery/components/DiscoveryAgentTable/DiscoveryAgentTable'),
  __esModule: true,
  default: () => {
    return <div className={'discovery-agent-table'}>Discovery Agent Table</div>
  }
}))

const paginationProps = {
  all: true,
  index: 0,
  totalItems: 1,
  totalPages: 1
}

const mockDiscoveryAgent: ApiGetAgentResponse = {
  accountIdentifier: '',
  id: '',
  identity: '',
  k8sConnector: {
    id: ''
  },
  name: '',
  organizationIdentifier: '',
  projectIdentifier: '',
  createdAt: '2023-06-21T13:17:41.491Z',
  createdBy: 'VgWXxi_6TdqAyplTQMg4CQ',
  updatedAt: '2023-06-21T13:17:41.491Z',
  installationDetails: {
    createdAt: '2023-06-21T13:17:41.491Z',
    createdBy: 'VgWXxi_6TdqAyplTQMg4CQ',
    updatedAt: '2023-06-21T13:17:41.491Z'
  },
  updatedBy: ''
}

const mockDiscoveryList = {
  items: [mockDiscoveryAgent],
  page: paginationProps
}

const fetchDiscoveryAgents = jest.fn(() => {
  return Object.create(mockDiscoveryList)
})

jest.mock('services/servicediscovery', () => ({
  useListAgent: jest.fn().mockImplementation(() => {
    return { data: mockDiscoveryList, refetch: fetchDiscoveryAgents, error: null, loading: false }
  })
}))

const PATH = routes.toDiscovery({ ...accountPathProps, ...projectPathProps, ...modulePathProps })
const PATH_PARAMS = {
  accountId: 'accountId',
  orgIdentifier: 'default',
  projectIdentifier: 'Discovery_Test',
  module: 'chaos'
}

describe('<DiscoveryPage /> tests', () => {
  beforeEach(() => jest.clearAllMocks())
  beforeAll(() => {
    jest.useFakeTimers().setSystemTime(new Date('2023-06-28'))
  })

  test('should match snapshot', async () => {
    const { container } = render(
      <TestWrapper path={PATH} pathParams={PATH_PARAMS}>
        <DiscoveryPage />
      </TestWrapper>
    )
    expect(useListAgent).toBeCalled()

    expect(container).toMatchSnapshot()
  })

  test('should render loading view correctly', async () => {
    jest.spyOn(servicediscovery, 'useListAgent').mockImplementation((): any => {
      return {
        data: undefined,
        loading: true
      }
    })

    const { container } = render(
      <TestWrapper path={PATH} pathParams={PATH_PARAMS}>
        <DiscoveryPage />
      </TestWrapper>
    )
    expect(useListAgent).toBeCalled()

    expect(container).toMatchSnapshot()
  })

  test('should render error view correctly', async () => {
    jest.spyOn(servicediscovery, 'useListAgent').mockImplementation((): any => {
      return {
        data: undefined,
        loading: false,
        error: {
          message: 'some error'
        }
      }
    })

    const { container, getByText } = render(
      <TestWrapper path={PATH} pathParams={PATH_PARAMS}>
        <DiscoveryPage />
      </TestWrapper>
    )
    expect(useListAgent).toBeCalled()

    expect(container).toMatchSnapshot()

    expect(getByText('Open Discovery Drawer')).toBeInTheDocument()
  })

  test('should render loaded view correctly', async () => {
    jest.spyOn(servicediscovery, 'useListAgent').mockImplementation((): any => {
      return {
        data: mockDiscoveryList,
        loading: false
      }
    })

    const { getByText } = render(
      <TestWrapper path={PATH} pathParams={PATH_PARAMS}>
        <DiscoveryPage />
      </TestWrapper>
    )
    expect(useListAgent).toBeCalled()

    expect(getByText('Discovery Agent Table')).toBeInTheDocument()
  })

  test('should render empty loaded view correctly', async () => {
    jest.spyOn(servicediscovery, 'useListAgent').mockImplementation((): any => {
      return {
        data: {
          items: [],
          page: paginationProps
        },
        loading: false
      }
    })

    const { getByText } = render(
      <TestWrapper path={PATH} pathParams={PATH_PARAMS}>
        <DiscoveryPage />
      </TestWrapper>
    )
    expect(useListAgent).toBeCalled()

    expect(getByText('Empty State Discovery Table')).toBeInTheDocument()
  })
})
