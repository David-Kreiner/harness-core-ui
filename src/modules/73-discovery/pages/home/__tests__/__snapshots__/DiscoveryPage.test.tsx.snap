// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<DiscoveryPage /> tests should match snapshot 1`] = `
<div>
  <div
    class="StyledProps--font StyledProps--main"
  >
    <div
      class="Layout--horizontal StyledProps--main PageHeader--container PageHeader--standard StyledProps--flex StyledProps--padding-left-xlarge StyledProps--padding-right-xlarge StyledProps--background-white"
      data-testid="page-header"
    >
      <div
        class="Layout--vertical StyledProps--main"
      >
        <div
          class="PageHeader--breadcrumbsDiv"
        >
          <div
            class="Layout--horizontal StyledProps--main Breadcrumbs--breadcrumbs StyledProps--flex StyledProps--flex-align-center-center StyledProps--flex-justifyContent-flex-start"
          >
            <div
              class="Layout--horizontal StyledProps--main StyledProps--flex StyledProps--flex-align-center-center StyledProps--flex-justifyContent-flex-start"
            >
              <a
                class="Breadcrumbs--breadcrumb"
                href="/account/accountId/chaos"
              >
                <span
                  class="bp3-icon StyledProps--main StyledProps--padding-right-xsmall"
                  data-icon="chaos-main"
                >
                  <svg
                    fill="none"
                    height="16"
                    viewBox="0 0 32 32"
                    width="16"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <path
                      clip-rule="evenodd"
                      d="M21.662 13.415a1.1 1.1 0 001.55 0 1.087 1.087 0 000-1.543l-5.647-5.62 6.837-3.929c1.6-.918 3.598.23 3.598 2.068v12.006l-8.265 1.968a1.09 1.09 0 10.51 2.122L28 18.641v8.968c0 1.838-1.999 2.986-3.598 2.068L14.89 24.21l1.331-9.568a1.092 1.092 0 00-.935-1.23 1.095 1.095 0 00-1.236.93l-1.209 8.69-8.642-4.966c-1.599-.919-1.599-3.216 0-4.134l11.4-6.551 6.063 6.033z"
                      fill="url(#chaos-main_svg__a)"
                      fill-rule="evenodd"
                    />
                    <defs>
                      <lineargradient
                        gradientUnits="userSpaceOnUse"
                        id="chaos-main_svg__a"
                        x1="12.755"
                        x2="27.888"
                        y1="6.364"
                        y2="29.992"
                      >
                        <stop
                          stop-color="#FF61A2"
                        />
                        <stop
                          offset="1"
                          stop-color="#D8005A"
                        />
                      </lineargradient>
                    </defs>
                  </svg>
                </span>
                <p
                  class="StyledProps--font StyledProps--main StyledProps--intent-primary StyledProps--font-size-small"
                >
                  Discovery_Test
                </p>
              </a>
              <span
                class="bp3-icon StyledProps--main StyledProps--color StyledProps--color-grey500 StyledProps--padding-right-xsmall StyledProps--padding-left-xsmall"
                data-icon="main-chevron-right"
              >
                <svg
                  height="8"
                  viewBox="0 0 8 14"
                  width="8"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M7.7 6.3l-6-6C1.5.1 1.3 0 1 0 .4 0 0 .4 0 1c0 .3.1.5.3.7L5.6 7 .3 12.3c-.2.2-.3.4-.3.7 0 .6.4 1 1 1 .3 0 .5-.1.7-.3l6-6c.2-.2.3-.4.3-.7s-.1-.5-.3-.7z"
                    fill="#4d5969"
                  />
                </svg>
              </span>
            </div>
          </div>
        </div>
        <div
          class="Layout--horizontal StyledProps--main"
        >
          <p
            class="StyledProps--font StyledProps--main StyledProps--font-variation-h4"
          >
            projectLabel common.discovery
          </p>
        </div>
      </div>
    </div>
    <div
      class="StyledProps--font StyledProps--main PageBody--pageBody"
      data-name="page-body"
      data-testid="page-body"
    >
      <div
        class="Layout--horizontal StyledProps--main PageSubHeader--container StyledProps--flex"
        data-testid="page-subheader"
      >
        <div
          class="Layout--horizontal StyledProps--main StyledProps--flex StyledProps--flex-justifyContent-space-between"
          style="width: 100%;"
        >
          <div
            class="Layout--horizontal StyledProps--main"
          >
            <button
              aria-label="discovery.homepage.newDiscoveryAgentBtn"
              class="bp3-button Button--button StyledProps--font StyledProps--main Button--with-current-color Button--withLeftIcon Button--variation Button--variation-primary"
              type="button"
            >
              <span
                class="bp3-icon bp3-icon-plus StyledProps--main StyledProps--padding-right-xsmall"
                icon="plus"
              >
                <svg
                  data-icon="plus"
                  height="16"
                  viewBox="0 0 16 16"
                  width="16"
                >
                  <desc>
                    plus
                  </desc>
                  <path
                    d="M13 7H9V3c0-.55-.45-1-1-1s-1 .45-1 1v4H3c-.55 0-1 .45-1 1s.45 1 1 1h4v4c0 .55.45 1 1 1s1-.45 1-1V9h4c.55 0 1-.45 1-1s-.45-1-1-1z"
                    fill-rule="evenodd"
                  />
                </svg>
              </span>
              <span
                class="bp3-button-text"
              >
                discovery.homepage.newDiscoveryAgentBtn
              </span>
            </button>
          </div>
          <div
            class="StyledProps--font StyledProps--main"
            data-name="monitoredServiceSeachContainer"
          >
            <div
              class="bp3-input-group ui-search-box ExpandingSearchInput--main   ExpandingSearchInput--alwaysExpanded ExpandingSearchInput--light"
              data-name=""
              style="width: 250px; max-width: 250px;"
            >
              <span
                class="bp3-icon StyledProps--main ExpandingSearchInput--icon StyledProps--color StyledProps--color-grey700"
                data-icon="thinner-search"
              >
                <svg
                  fill="none"
                  height="14"
                  viewBox="0 0 16 16"
                  width="14"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M7.125 13.25c1.617 0 3.32-.695 4.375-1.75L15 15l-3.5-3.5c1.021-1.005 1.75-2.894 1.75-4.375a6.125 6.125 0 10-6.125 6.125z"
                    stroke="currentColor"
                    stroke-linecap="round"
                    stroke-linejoin="round"
                  />
                </svg>
              </span>
              <input
                class="bp3-input  ExpandingSearchInput--filter"
                dir="auto"
                placeholder="discovery.homepage.searchDiscoveryAgent"
                style="padding-right: 38px;"
                type="search"
                value=""
              />
            </div>
          </div>
        </div>
      </div>
      <div
        class="StyledProps--font StyledProps--main PageBody--pageBody discoveryAgentTable"
        data-name="page-body"
        data-testid="page-body"
      >
        <div
          class="discovery-agent-table"
        >
          Discovery Agent Table
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`<DiscoveryPage /> tests should render error view correctly 1`] = `
<div>
  <div
    class="StyledProps--font StyledProps--main"
  >
    <div
      class="Layout--horizontal StyledProps--main PageHeader--container PageHeader--standard StyledProps--flex StyledProps--padding-left-xlarge StyledProps--padding-right-xlarge StyledProps--background-white"
      data-testid="page-header"
    >
      <div
        class="Layout--vertical StyledProps--main"
      >
        <div
          class="PageHeader--breadcrumbsDiv"
        >
          <div
            class="Layout--horizontal StyledProps--main Breadcrumbs--breadcrumbs StyledProps--flex StyledProps--flex-align-center-center StyledProps--flex-justifyContent-flex-start"
          >
            <div
              class="Layout--horizontal StyledProps--main StyledProps--flex StyledProps--flex-align-center-center StyledProps--flex-justifyContent-flex-start"
            >
              <a
                class="Breadcrumbs--breadcrumb"
                href="/account/accountId/chaos"
              >
                <span
                  class="bp3-icon StyledProps--main StyledProps--padding-right-xsmall"
                  data-icon="chaos-main"
                >
                  <svg
                    fill="none"
                    height="16"
                    viewBox="0 0 32 32"
                    width="16"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <path
                      clip-rule="evenodd"
                      d="M21.662 13.415a1.1 1.1 0 001.55 0 1.087 1.087 0 000-1.543l-5.647-5.62 6.837-3.929c1.6-.918 3.598.23 3.598 2.068v12.006l-8.265 1.968a1.09 1.09 0 10.51 2.122L28 18.641v8.968c0 1.838-1.999 2.986-3.598 2.068L14.89 24.21l1.331-9.568a1.092 1.092 0 00-.935-1.23 1.095 1.095 0 00-1.236.93l-1.209 8.69-8.642-4.966c-1.599-.919-1.599-3.216 0-4.134l11.4-6.551 6.063 6.033z"
                      fill="url(#chaos-main_svg__a)"
                      fill-rule="evenodd"
                    />
                    <defs>
                      <lineargradient
                        gradientUnits="userSpaceOnUse"
                        id="chaos-main_svg__a"
                        x1="12.755"
                        x2="27.888"
                        y1="6.364"
                        y2="29.992"
                      >
                        <stop
                          stop-color="#FF61A2"
                        />
                        <stop
                          offset="1"
                          stop-color="#D8005A"
                        />
                      </lineargradient>
                    </defs>
                  </svg>
                </span>
                <p
                  class="StyledProps--font StyledProps--main StyledProps--intent-primary StyledProps--font-size-small"
                >
                  Discovery_Test
                </p>
              </a>
              <span
                class="bp3-icon StyledProps--main StyledProps--color StyledProps--color-grey500 StyledProps--padding-right-xsmall StyledProps--padding-left-xsmall"
                data-icon="main-chevron-right"
              >
                <svg
                  height="8"
                  viewBox="0 0 8 14"
                  width="8"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M7.7 6.3l-6-6C1.5.1 1.3 0 1 0 .4 0 0 .4 0 1c0 .3.1.5.3.7L5.6 7 .3 12.3c-.2.2-.3.4-.3.7 0 .6.4 1 1 1 .3 0 .5-.1.7-.3l6-6c.2-.2.3-.4.3-.7s-.1-.5-.3-.7z"
                    fill="#4d5969"
                  />
                </svg>
              </span>
            </div>
          </div>
        </div>
        <div
          class="Layout--horizontal StyledProps--main"
        >
          <p
            class="StyledProps--font StyledProps--main StyledProps--font-variation-h4"
          >
            projectLabel common.discovery
          </p>
        </div>
      </div>
    </div>
    <div
      class="StyledProps--font StyledProps--main PageBody--pageBody"
      data-name="page-body"
      data-testid="page-body"
    >
      <div
        class="discovery-view-mock"
      >
        <h1>
          Empty State Discovery Table
        </h1>
        <button>
          Open Discovery Drawer
        </button>
      </div>
    </div>
  </div>
</div>
`;

exports[`<DiscoveryPage /> tests should render loading view correctly 1`] = `
<div>
  <div
    class="StyledProps--font StyledProps--main"
  >
    <div
      class="Layout--horizontal StyledProps--main PageHeader--container PageHeader--standard StyledProps--flex StyledProps--padding-left-xlarge StyledProps--padding-right-xlarge StyledProps--background-white"
      data-testid="page-header"
    >
      <div
        class="Layout--vertical StyledProps--main"
      >
        <div
          class="PageHeader--breadcrumbsDiv"
        >
          <div
            class="Layout--horizontal StyledProps--main Breadcrumbs--breadcrumbs StyledProps--flex StyledProps--flex-align-center-center StyledProps--flex-justifyContent-flex-start"
          >
            <div
              class="Layout--horizontal StyledProps--main StyledProps--flex StyledProps--flex-align-center-center StyledProps--flex-justifyContent-flex-start"
            >
              <a
                class="Breadcrumbs--breadcrumb"
                href="/account/accountId/chaos"
              >
                <span
                  class="bp3-icon StyledProps--main StyledProps--padding-right-xsmall"
                  data-icon="chaos-main"
                >
                  <svg
                    fill="none"
                    height="16"
                    viewBox="0 0 32 32"
                    width="16"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <path
                      clip-rule="evenodd"
                      d="M21.662 13.415a1.1 1.1 0 001.55 0 1.087 1.087 0 000-1.543l-5.647-5.62 6.837-3.929c1.6-.918 3.598.23 3.598 2.068v12.006l-8.265 1.968a1.09 1.09 0 10.51 2.122L28 18.641v8.968c0 1.838-1.999 2.986-3.598 2.068L14.89 24.21l1.331-9.568a1.092 1.092 0 00-.935-1.23 1.095 1.095 0 00-1.236.93l-1.209 8.69-8.642-4.966c-1.599-.919-1.599-3.216 0-4.134l11.4-6.551 6.063 6.033z"
                      fill="url(#chaos-main_svg__a)"
                      fill-rule="evenodd"
                    />
                    <defs>
                      <lineargradient
                        gradientUnits="userSpaceOnUse"
                        id="chaos-main_svg__a"
                        x1="12.755"
                        x2="27.888"
                        y1="6.364"
                        y2="29.992"
                      >
                        <stop
                          stop-color="#FF61A2"
                        />
                        <stop
                          offset="1"
                          stop-color="#D8005A"
                        />
                      </lineargradient>
                    </defs>
                  </svg>
                </span>
                <p
                  class="StyledProps--font StyledProps--main StyledProps--intent-primary StyledProps--font-size-small"
                >
                  Discovery_Test
                </p>
              </a>
              <span
                class="bp3-icon StyledProps--main StyledProps--color StyledProps--color-grey500 StyledProps--padding-right-xsmall StyledProps--padding-left-xsmall"
                data-icon="main-chevron-right"
              >
                <svg
                  height="8"
                  viewBox="0 0 8 14"
                  width="8"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M7.7 6.3l-6-6C1.5.1 1.3 0 1 0 .4 0 0 .4 0 1c0 .3.1.5.3.7L5.6 7 .3 12.3c-.2.2-.3.4-.3.7 0 .6.4 1 1 1 .3 0 .5-.1.7-.3l6-6c.2-.2.3-.4.3-.7s-.1-.5-.3-.7z"
                    fill="#4d5969"
                  />
                </svg>
              </span>
            </div>
          </div>
        </div>
        <div
          class="Layout--horizontal StyledProps--main"
        >
          <p
            class="StyledProps--font StyledProps--main StyledProps--font-variation-h4"
          >
            projectLabel common.discovery
          </p>
        </div>
      </div>
    </div>
    <div
      class="StyledProps--font StyledProps--main PageBody--pageBody"
      data-name="page-body"
      data-testid="page-body"
    >
      <div
        class="StyledProps--font StyledProps--main PageSpinner--spinner StyledProps--flex StyledProps--flex-align-center-center"
        data-testid="page-spinner"
      >
        <div
          class="Layout--vertical Layout--layout-spacing-medium StyledProps--main PageSpinner--content"
          style="align-items: center; width: 500px;"
        >
          <span
            class="bp3-icon StyledProps--main StyledProps--color StyledProps--color-grey600"
            data-icon="steps-spinner"
          >
            <svg
              height="32"
              viewBox="0 0 128 128"
              width="32"
              xmlns="http://www.w3.org/2000/svg"
            >
              <g>
                <path
                  d="M122.5 69.25H96.47a33.1 33.1 0 000-10.5h26.05a5.25 5.25 0 010 10.5z"
                  fill="#ff8e02"
                />
                <path
                  d="M112.04 97.83L89.47 84.8a33.1 33.1 0 005.25-9.1l22.57 13.03a5.25 5.25 0 01-5.28 9.1zM88.68 117.35L75.65 94.78a33.1 33.1 0 009.1-5.25l13.02 22.57a5.25 5.25 0 11-9.1 5.25zM58.7 122.57V96.5a33.1 33.1 0 0010.5 0v26.07a5.25 5.25 0 01-10.5 0zM30.1 112.1l13.04-22.57a33.1 33.1 0 009.1 5.25L39.2 117.35a5.25 5.25 0 11-9.1-5.25zM10.6 88.74L33.16 75.7a33.1 33.1 0 005.25 9.1L15.88 97.83a5.25 5.25 0 11-5.25-9.1z"
                  fill="#ffddb3"
                  fill-opacity="0.3"
                />
                <path
                  d="M5.37 58.75h26.06a33.1 33.1 0 000 10.5H5.37a5.25 5.25 0 010-10.5z"
                  fill="#ffd29a"
                  fill-opacity="0.4"
                />
                <path
                  d="M15.85 30.17L38.4 43.2a33.1 33.1 0 00-5.24 9.1L10.6 39.25a5.25 5.25 0 115.25-9.1z"
                  fill="#ffc680"
                  fill-opacity="0.5"
                />
                <path
                  d="M39.2 10.65l13.03 22.57a33.1 33.1 0 00-9.1 5.25l-13-22.57a5.25 5.25 0 119.1-5.25z"
                  fill="#ffbb67"
                  fill-opacity="0.6"
                />
                <path
                  d="M69.2 5.43V31.5a33.1 33.1 0 00-10.5 0V5.42a5.25 5.25 0 1110.5 0z"
                  fill="#ffb04d"
                  fill-opacity="0.7"
                />
                <path
                  d="M97.77 15.9L84.75 38.47a33.1 33.1 0 00-9.1-5.25l13.03-22.57a5.25 5.25 0 119.1 5.25z"
                  fill="#ffa535"
                  fill-opacity="0.8"
                />
                <path
                  d="M117.3 39.26L94.7 52.3a33.1 33.1 0 00-5.25-9.1l22.57-13.03a5.25 5.25 0 015.25 9.1z"
                  fill="#ff991b"
                  fill-opacity="0.9"
                />
                <animatetransform
                  attributeName="transform"
                  calcMode="discrete"
                  dur="1200ms"
                  repeatCount="indefinite"
                  type="rotate"
                  values="0 64 64;30 64 64;60 64 64;90 64 64;120 64 64;150 64 64;180 64 64;210 64 64;240 64 64;270 64 64;300 64 64;330 64 64"
                />
              </g>
            </svg>
          </span>
          <p
            class="StyledProps--font StyledProps--main StyledProps--font-size-medium StyledProps--font-align-center StyledProps--color StyledProps--color-grey600"
          >
            Loading, please wait...
          </p>
        </div>
      </div>
      <div
        class="discovery-view-mock"
      >
        <h1>
          Empty State Discovery Table
        </h1>
        <button>
          Open Discovery Drawer
        </button>
      </div>
    </div>
  </div>
</div>
`;
